internal/policy/verify.go

```diff
@@ -643,3 +649,46 @@ func getChangedPaths(repo *git.Repository, entry *rsl.ReferenceEntry) ([]string,
 
 	return gitinterface.GetDiffFilePaths(currentCommit, priorCommit)
 }
+
+func checkCommitAgainstModifiedPaths(ctx context.Context, repo *git.Repository, commit *object.Commit, commitPolicy *State) ([]string, error) {
+	filePaths, err := gitinterface.GetFilePathsChangedByCommit(repo, commit)
+	if err != nil {
+		return []string{}, err
+	}
+
+	badPaths := []string{}
+	for _, path := range filePaths {
+		keys, err := commitPolicy.FindPublicKeysForPath(ctx, fmt.Sprintf("file:%s", path))
+		if err != nil {
+			return []string{}, err
+		}
+
+		// Verify the keys associated with the commit. If verification fails, add the path to the list of bad paths.
+		err = verifyKeys(ctx, commit, keys)
+
+		if err != nil {
+			if strings.HasPrefix(err.Error(), "good signature from key") {
+				continue
+			}
+			badPaths = append(badPaths, path)
+		}
+	}
+	return badPaths, nil
+}
+
+func verifyKeys(ctx context.Context, commit *object.Commit, keys []*tuf.Key) error {
+	for _, key := range keys {
+		err := gitinterface.VerifyCommitSignature(ctx, commit, key)
+		if err == nil {
+			return fmt.Errorf(goodSignatureMessageFmt, key.KeyType, key.KeyID)
+		}
+
+		if errors.Is(err, gitinterface.ErrUnknownSigningMethod) {
+			// We encounter this for key types that can be used for gittuf
+			// policy metadata but not Git objects
+			continue
+		}
```

Comment:
You want to handle a case where it's not `ErrIncorrectVerificationKey` after this for cases where you have an unexpected error from `VerifyCommitSignature`.

Reply:
Handled